### PR TITLE
[FLINK-36709][parquet] fix parquet can not read row with last column is array.

### PR DESCRIPTION
**Pull request by Stephen0421**
## What is the purpose of the change

*(For example: This pull request makes task deployment go through the blob server, rather than through RPC. That way we avoid re-transferring them on each deployment (during recovery).)*
Fix parquet can not read row with last column is array.

## Brief change log

*(for example:)*
  - fix a corner condition.

## Verifying this change

Please make sure both new and modified tests in this PR follow [the conventions for tests defined in our code quality guide](https://flink.apache.org/how-to-contribute/code-style-and-quality-common/#7-testing).

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

  - modify unit test case to cover the situation.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper:no
  - The S3 file system connector:no

## Documentation

  - Does this pull request introduce a new feature? no

### CODE REVIEW

File: flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/utils/NestedPositionUtil.java [hunk 1]
```diff
@@ -50,12 +50,11 @@ public static RowPosition calculateRowOffsets(
         int nullValuesCount = 0;
         BooleanArrayList nullRowFlags = new BooleanArrayList(0);
         for (int i = 0; i < fieldDefinitionLevels.length; i++) {
+            // If row has array type field, the field repetition level's count will bigger than the
+            // actual row value count, so we need to skip those value that bigger than row's
+            // repetition level.
```
Comment (xccui) [inline flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/utils/NestedPositionUtil.java:55]:
```suggestion
            // If a row's last field is an array, the repetition levels for the array's items will be
            // larger than the parent row's repetition level, so we need to skip those values.
```

**Review by xccui — APPROVED** [review]
Tested the change locally. It worked well! Thanks!

File: flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/utils/NestedPositionUtil.java [hunk 2]
```diff
@@ -50,12 +50,10 @@ public static RowPosition calculateRowOffsets(
         int nullValuesCount = 0;
         BooleanArrayList nullRowFlags = new BooleanArrayList(0);
         for (int i = 0; i < fieldDefinitionLevels.length; i++) {
+            // If a row's last field is an array, the repetition levels for the array's items will be
+            // larger than the parent row's repetition level, so we need to skip those values.
```
Comment (prm-xingcan) [inline flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/utils/NestedPositionUtil.java:54]: A lint issue. Sorry.
```suggestion
            // If a row's last field is an array, the repetition levels for the array's items will
            // be larger than the parent row's repetition level, so we need to skip those values.
```

Comment (xccui) [inline flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/utils/NestedPositionUtil.java:54]: A lint issue. Sorry.
```suggestion
            // If a row's last field is an array, the repetition levels for the array's items will
            // be larger than the parent row's repetition level, so we need to skip those values.
```

**Comment by xccui** [comment]
@flinkbot run azure

**Pull request merged**